platforms/jvm/language-java/src/integTest/groovy/org/gradle/api/tasks/compile/JavaCompileProblemsIntegrationTest.groovy

```diff
@@ -229,24 +244,28 @@ class JavaCompileProblemsIntegrationTest extends AbstractIntegrationSpec impleme
 
         then:
         verifyAll(receivedProblem(0)) {
-            assertProblem(it, "ERROR", true)
-            fqid == 'compilation:java:java-compilation-error'
-            details == '\';\' expected'
+            assertLocations(it, true)
+            severity == Severity.ERROR
+            fqid == 'compilation:java:compiler-err-expected'
+            contextualLabel == '\';\' expected'
         }
         verifyAll(receivedProblem(1)) {
-            assertProblem(it, "ERROR", true)
-            fqid == 'compilation:java:java-compilation-error'
-            details == '\';\' expected'
+            assertLocations(it, true)
+            severity == Severity.ERROR
+            fqid == 'compilation:java:compiler-err-expected'
+            contextualLabel == '\';\' expected'
         }
         verifyAll(receivedProblem(2)) {
-            assertProblem(it, "WARNING", true)
-            fqid == 'compilation:java:java-compilation-warning'
-            details == 'redundant cast to java.lang.String'
+            assertLocations(it, true)
+            severity == Severity.WARNING
+            fqid == 'compilation:java:compiler-warn-redundant-cast'
+            contextualLabel == 'redundant cast to java.lang.String'
         }
         verifyAll(receivedProblem(3)) {
-            assertProblem(it, "WARNING", true)
-            fqid == 'compilation:java:java-compilation-warning'
-            details == 'redundant cast to java.lang.String'
+            assertLocations(it, true)
```

Comment:
I was never a fan of boolean parameters for functions. Instead, It's not strictly related to this pr, but I'd recommend to introduce a new `assertPreciseLocation` method.

Reply:
I would not force it into this PR, but we could make an issue to review the structure there.